packages/catalog-client/src/types.ts

```diff
@@ -25,15 +25,32 @@ export type CatalogListResponse<T> = {
   items: T[];
 };
 
+export type ApiContext = {
```

Comment:
Think we should just call this `RequestOptions` tbh, or maybe we can just avoid exporting this type for now?

Reply:
I can rename it. Note though that I just renamed it from CatalogRequestOptions to [Api]Context as per @freben wishes? Please feel free to take this PR the last mile, it will likely go faster if you bypass me ;)

There are a few files that use the CatalogRequestOptions type. Would you rather duplicate the type a few times or export/import?

Reply:
Ah yeah we talked about adding a second argument with context, I think it's have to cover options as well for it to be implemented in a forwards compatible way. If we want to add more options to any of the methods in the future we wouldn't want to have that as a 3rd argument imo, `fn(id, ctx, opts)` is a bit backwards imo, `fn(id, ops, ctx)` I think would work, or in this case `fn(id, opts & ctx)`. I mean tbh what's really the difference between context and options here? 😅 

I don't see the naming request from freben though, he suggested `CatalogRequestOptions` above.

Reply:
And alright will have a look at getting this across the line